src/videojs-hls.js

```diff
@@ -944,41 +1013,32 @@ videojs.Hls.prototype.drainBuffer = function(event) {
 
   event = event || {};
 
+  if (segmentInfo.mediaIndex > 0) {
+    segmentTimestampOffset = videojs.Hls.Playlist.duration(segmentInfo.playlist, segmentInfo.mediaIndex);
+  }
+
   // If we have seeked into a non-buffered time-range, remove all buffered
   // time-ranges because they could have been incorrectly placed originally
   if (this.tech_.seeking() && outsideBufferedRanges) {
-    if (hasBufferedContent) {
-      // In Chrome, it seems that too many independent buffered time-ranges can
-      // cause playback to fail to resume when seeking so just kill all of them
-      this.sourceBuffer.remove(0, Infinity);
-      return;
-    }
-
     // If there are discontinuities in the playlist, we can't be sure of anything
     // related to time so we reset the timestamp offset and start appending data
     // anew on every seek
     if (segmentInfo.playlist.discontinuityStarts.length) {
-      if (segmentInfo.mediaIndex > 0) {
-        segmentTimestampOffset = videojs.Hls.Playlist.duration(segmentInfo.playlist, segmentInfo.mediaIndex);
-      }
-
       // Now that the forward buffer is clear, we have to set timestamp offset to
       // the start of the buffered region
       this.sourceBuffer.timestampOffset = segmentTimestampOffset;
     }
-  } else if (segment.discontinuity) {
+  } else if (segment.discontinuity && currentBuffered.length) {
     // If we aren't seeking and are crossing a discontinuity, we should set
     // timestampOffset for new segments to be appended the end of the current
     // buffered time-range
     this.sourceBuffer.timestampOffset = currentBuffered.end(0);
+  } else if (!hasBufferedContent && this.tech_.currentTime() > 0.05) {
```

Comment:
No this is to cover the strange condition mentioned in the comment below. We are not seeking, current time is not zero and we don't have any buffered data. In that case we need to set `timestampOffset` before placing the segment. I saw this several times while trying to get live streams to work.